batch/batch/server/server.py

```diff
@@ -175,8 +175,13 @@ def parent_new_state(self, new_state, parent_id, maybe_exit_code):
             log.info(f'parent {parent_id} successfully complete for {self.id}')
             self.incomplete_parent_ids.discard(parent_id)
             if not self.incomplete_parent_ids:
-                log.info(f'all parents successfully complete for {self.id}')
-                self._create_pod()
+                if self._state != 'Cancelled':
```

Comment:
Would ‘self.is_complete()’ catch any additional cases (meaning spawning child after parent previously marked complete due to a logical race, or some network retry)?

Reply:
There are four k8s states: `Created`, `Cancelled`, `Complete` (annoyingly not a verb), and `Running`. If we're in `Complete` or `Running` here, that's a bug. I've added an assert